zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/EphemeralNodeThrottlingTest.java

```diff
@@ -0,0 +1,201 @@
+            leaderServer.create(PATH + "-leader-", new byte[512], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL_SEQUENTIAL);
+            followerServer.create(PATH + "-follower-", new byte[512], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL_SEQUENTIAL);
+        }
+
+        try {
+            leaderServer.create(PATH + "-leader-", new byte[512], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL_SEQUENTIAL);
+        } catch (KeeperException.EphemeralCountExceededException e) {
+            leaderThrewError = true;
+        }
+        try {
+            followerServer.create(PATH + "-follower-", new byte[512], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL_SEQUENTIAL);
+        } catch (KeeperException.EphemeralCountExceededException e) {
+            followerThrewError = true;
+        }
+        assertTrue(leaderThrewError && followerThrewError);
+
+        long actual = (long) MetricsUtils.currentServerMetrics().get("ephemeral_node_max_count_violation");
+        assertEquals(2, actual);
+
+        servers.shutDownAllServers();
+    }
+
+    @Test
+    public void rejectedEphemeralCreatesMetricsTest() throws Exception {
+        int ephemeralExcess = DEFAULT_MAX_EPHEMERAL_NODES / 2;
+        System.setProperty("zookeeper.ephemeral.count.limit", Integer.toString(DEFAULT_MAX_EPHEMERAL_NODES));
+        servers = LaunchServers(NUM_SERVERS);
+        for (int i = 0; i < DEFAULT_MAX_EPHEMERAL_NODES + ephemeralExcess; i++) {
+            try {
+                servers.zk[0].create(PATH + "-" + i, new byte[512], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL);
+            } catch (KeeperException.EphemeralCountExceededException e) {
+                LOG.info("Encountered EphemeralCountExceededException as expected, continuing...");
+            }
+        }
+
+        long actual = (long) MetricsUtils.currentServerMetrics().get("ephemeral_node_max_count_violation");
+        assertEquals(ephemeralExcess, actual);
+
+        servers.shutDownAllServers();
+    }
+
+    // Tests multithreaded creates and deletes against the leader and a follower server
+    @Test
+    public void multithreadedRequestsTest() throws Exception {
+        int ephemeralNodeLimit = 7500;
+        System.setProperty("zookeeper.ephemeral.count.limit", Integer.toString(ephemeralNodeLimit));
+
+        servers = LaunchServers(NUM_SERVERS);
+        ZooKeeper leaderServer = servers.zk[servers.findLeader()];
+        ZooKeeper followerServer = servers.zk[servers.findAnyFollower()];
+
+        runMultithreadedRequests(leaderServer);
+        runMultithreadedRequests(followerServer);
+
+        // TODO: What % delta do we want to allow here?
+        assertEquals(ephemeralNodeLimit, leaderServer.getEphemerals().size(), ephemeralNodeLimit/20d);
+        assertEquals(ephemeralNodeLimit, followerServer.getEphemerals().size(), ephemeralNodeLimit/20d);
+
+        servers.shutDownAllServers();
+    }
+
+    private void runMultithreadedRequests(ZooKeeper server) {
```

Comment:
nice! nit: We can pass servers array to runMultithreadedRequests method and randomly pick leaderServer or followerServer on every create.